### ISSUE TITLE
Build error at ResolveRhetosProjectAssets: FileNotFoundException Newtonsoft.Json

### ISSUE THREAD

**Issue by bantolov** [Rhetos/Rhetos#432]
**Describe the bug**
When building an application with Rhetos framework v4 or v5 fails with error
`The "ResolveRhetosProjectAssets" task failed unexpectedly.`
`System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The system cannot find the file specified.`

**To Reproduce**
Unfortunately, the error occurs sporadically, I could not find a cause.
On a same project it behaves differently on different machines.
I'm currently using Visual Studio Enterprise 2019 v16.9.3.

It is interesting that the **build works in dotnet CLI**:

* `msbuild CommonConceptsTest.sln` => ERROR
* Building CommonConceptsTest.sln in Visual Studio => ERROR
* `dotnet build CommonConceptsTest.sln` => works well

I am currently able to reproduce the error on CommonConceptsTest.sln solution in [Rhetos](https://github.com/Rhetos/Rhetos), commit 571cffc:
`msbuild CommonConceptsTest.sln`

**Error message**
Build output:
```log
Build started...
1>------ Build started: Project: CommonConcepts.Test, Configuration: Debug Any CPU ------
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018: The "ResolveRhetosProjectAssets" task failed unexpectedly.
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018: System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The system cannot find the file specified.
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018: File name: 'Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:    at NuGet.ProjectModel.LockFileFormat.Read(TextReader reader, ILogger log, String path)
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:    at NuGet.ProjectModel.LockFileFormat.Read(Stream stream, ILogger log, String path)
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:    at NuGet.ProjectModel.LockFileUtilities.<>c__DisplayClass0_0.<GetLockFile>b__0(FileStream stream, String path)
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:    at NuGet.Common.FileUtility.SafeRead[T](String filePath, Func`3 read)
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:    at NuGet.ProjectModel.LockFileUtilities.GetLockFile(String lockFilePath, ILogger logger)
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:    at Rhetos.NuGetUtilities..ctor(String projectRootFolder, IEnumerable`1 projectContentFiles, NuGetLogger logger, String target) in C:\My Projects\Rhetos5\Source\RhetosVSIntegration\NuGetUtilities.cs:line 52
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:    at RhetosVSIntegration.ResolveRhetosProjectAssets.Execute() in C:\My Projects\Rhetos5\Source\RhetosVSIntegration\ResolveRhetosProjectAssets.cs:line 68
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018: 
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018: WRN: Assembly binding logging is turned OFF.
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018: To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018: Note: There is some performance penalty associated with assembly bind failure logging.
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018: To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
1>C:\Users\bantolovic\.nuget\packages\rhetos.msbuild\5.0.0-dev2103231357f23d39\build\Rhetos.MSBuild.targets(19,9): error MSB4018:
1>Done building project "CommonConcepts.Test.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

**Comment by bantolov** [comment]
The issue might be caused by the custom MSBuild task ResolveRhetosProjectAssets, that targets .NET Standard 2.0.

There are multiple discussions that mention issues with dependencies when using .NET Standard in custom task ([1](https://github.com/AArnott/CodeGeneration.Roslyn/issues/57#issuecomment-353982858), [2](https://github.com/dotnet/msbuild/issues/2111#issuecomment-658386764)).
It seems that the only reliable solution is multi-targeting for .NET Core and .NET Framework, to suport both `msbuild` (visual studio) and `dotnet build`.

**Comment by bantolov** [comment]
A temporary workaround is to add the following `assemblyBinding` element in *MSBuild.exe.config* on the development machine.

```xml
      <dependentAssembly>
        <assemblyIdentity name="Autofac" publicKeyToken="17863af14b0044da" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.9.4.0" newVersion="4.9.4.0" />
      </dependentAssembly>
```

You can locate the .config file by running `where msbuild` in Developer Command Prompt.

**Comment by bantolov** [comment]
The issue is solved in Rhetos v4.4 and Rhetos v5.0.

ResolveRhetosProjectAssets MSBuild task now uses custom assembly resolver that ignores the assembly versions (similar effect to assemblyBinding configuration above).
